Dependabot fails to determine NuGet dependencies due to shallow clone and GitVersion

For this repository, Dependabot updates for NuGet packages always emits errors such as follows, even though the CI succeeded:

[nuget in /. - Update #1074132945 - Dependabot Updates #112](https://github.com/y-iihoshi/ThScoreFileConverter/actions/runs/16914398911)
CI Log: https://github.com/y-iihoshi/ThScoreFileConverter/actions/runs/16914398911/job/47924452673#step:3:1582
```
updater | 2025/08/12 16:10:00 WARN   Error determining dependencies from `/home/dependabot/dependabot-updater/repo/TemplateGenerator/TemplateGenerator.csproj`:
STDOUT:
  Determining projects to restore...
  Restored /home/dependabot/dependabot-updater/repo/ThScoreFileConverter.Core/ThScoreFileConverter.Core.csproj (in 4.38 sec).
  Restored /home/dependabot/dependabot-updater/repo/TemplateGenerator/TemplateGenerator.csproj (in 7.95 sec).
  INFO [08/12/25 16:10:00:55] Applicable build agent found: 'GitHubActions'.
  INFO [08/12/25 16:10:00:59] Working directory: /home/dependabot/dependabot-updater/repo/ThScoreFileConverter.Core
  INFO [08/12/25 16:10:00:61] Project root is: /home/dependabot/dependabot-updater/repo/
  INFO [08/12/25 16:10:00:61] DotGit directory is: /home/dependabot/dependabot-updater/repo/.git
  INFO [08/12/25 16:10:00:61] Branch from build environment: 
  INFO [08/12/25 16:10:00:61] Begin: Normalizing git directory for branch ''
    INFO [08/12/25 16:10:00:63] One remote found (origin -> 'https://github.com/y-iihoshi/ThScoreFileConverter').
    INFO [08/12/25 16:10:00:63] Adding refspec: +refs/heads/*:refs/remotes/origin/*
    INFO [08/12/25 16:10:00:64] Skipping fetching, if GitVersion does not calculate your version as expected you might need to allow fetching or use dynamic repositories
    INFO [08/12/25 16:10:00:65] HEAD points at branch 'refs/heads/main'.
    INFO [08/12/25 16:10:00:65] End: Normalizing git directory for branch '' (Took: 42.53ms)
    INFO [08/12/25 16:10:00:66] Begin: Loading version variables from disk cache
      INFO [08/12/25 16:10:00:66] Cache file /home/dependabot/dependabot-updater/repo/.git/gitversion_cache/C6F362EE0B55DE4719AE6197F1F4948CC4C3F151.yml not found.
    INFO [08/12/25 16:10:00:66] End: Loading version variables from disk cache (Took: 0.37ms)
    INFO [08/12/25 16:10:00:66] Using latest commit on specified branch
    INFO [08/12/25 16:10:00:70] Running against branch: main (fca4cec Remove BOM to avoid parsing error)
    INFO [08/12/25 16:10:00:70] Begin: Calculating base versions
    INFO [08/12/25 16:10:00:72] End: Calculating base versions (Took: 13.63ms)
    ERROR [08/12/25 16:10:00:73] An unexpected error occurred:
  GitVersion.GitVersionException: Cannot find commit fca4cec Remove BOM to avoid parsing error. Please ensure that the repository is an unshallow clone with `git fetch --unshallow`.
   ---> LibGit2Sharp.NotFoundException: object not found - no match for id (feac0b63a7c66b99965759b10b9e3d9ab4feeed2)
     at LibGit2Sharp.Core.Ensure.HandleError(Int32 result) in /_/LibGit2Sharp/Core/Ensure.cs:line 154
     at LibGit2Sharp.Core.Ensure.ZeroResult(Int32 result) in /_/LibGit2Sharp/Core/Ensure.cs:line 172
     at LibGit2Sharp.Core.Proxy.git_revwalk_next(RevWalkerHandle walker) in /_/LibGit2Sharp/Core/Proxy.cs:line 2774
     at LibGit2Sharp.CommitLog.CommitEnumerator.MoveNext() in /_/LibGit2Sharp/CommitLog.cs:line 140
     at System.Linq.Enumerable.SelectEnumerableIterator`2.MoveNext()
     at System.Linq.Enumerable.TryGetFirst[TSource](IEnumerable`1 source, Func`2 predicate, Boolean& found)
     at System.Linq.Enumerable.First[TSource](IEnumerable`1 source, Func`2 predicate)
     at GitVersion.RepositoryStore.GetBaseVersionSource(ICommit currentBranchTip) in D:\a\GitVersion\GitVersion\src\GitVersion.Core\Core\RepositoryStore.cs:line 64
     --- End of inner exception stack trace ---
     at GitVersion.RepositoryStore.GetBaseVersionSource(ICommit currentBranchTip) in D:\a\GitVersion\GitVersion\src\GitVersion.Core\Core\RepositoryStore.cs:line 68
     at GitVersion.VersionCalculation.FallbackVersionStrategy.GetBaseVersions(EffectiveBranchConfiguration configuration)+MoveNext() in D:\a\GitVersion\GitVersion\src\GitVersion.Core\VersionCalculation\BaseVersionCalculators\FallbackVersionStrategy.cs:line 25
     at GitVersion.VersionCalculation.BaseVersionCalculator.GetBaseVersions(IVersionStrategy strategy, EffectiveBranchConfiguration configuration)+MoveNext() in D:\a\GitVersion\GitVersion\src\GitVersion.Core\VersionCalculation\BaseVersionCalculator.cs:line 132
     at System.Collections.Generic.List`1..ctor(IEnumerable`1 collection)
     at System.Linq.Enumerable.ToList[TSource](IEnumerable`1 source)
     at GitVersion.VersionCalculation.BaseVersionCalculator.GetBaseVersion() in D:\a\GitVersion\GitVersion\src\GitVersion.Core\VersionCalculation\BaseVersionCalculator.cs:line 48
     at GitVersion.VersionCalculation.NextVersionCalculator.FindVersion() in D:\a\GitVersion\GitVersion\src\GitVersion.Core\VersionCalculation\NextVersionCalculator.cs:line 56
     at GitVersion.GitVersionCalculateTool.CalculateVersionVariables() in D:\a\GitVersion\GitVersion\src\GitVersion.Core\Core\GitVersionCalculateTool.cs:line 52
     at GitVersion.GitVersionExecutor.RunGitVersionTool(GitVersionOptions gitVersionOptions) in D:\a\GitVersion\GitVersion\src\GitVersion.App\GitVersionExecutor.cs:line 66
    INFO [08/12/25 16:10:00:73] Attempting to show the current git graph (please include in issue): 
    INFO [08/12/25 16:10:00:73] Showing max of 100 commits
    INFO [08/12/25 16:10:00:77] * fca4cec 2 minutes ago  (grafted, HEAD -> main, origin/main, origin/HEAD)
  
/home/dependabot/.nuget/packages/gitversion.msbuild/5.12.0/tools/GitVersion.MsBuild.targets(9,9): error MSB3073: The command "dotnet --roll-forward Major "/home/dependabot/.nuget/packages/gitversion.msbuild/5.12.0/tools/netcoreapp3.1/gitversion.dll" "/home/dependabot/dependabot-updater/repo/ThScoreFileConverter.Core"  -output file -outputfile obj/gitversion.json" exited with code 1. [/home/dependabot/dependabot-updater/repo/ThScoreFileConverter.Core/ThScoreFileConverter.Core.csproj::TargetFramework=net8.0-windows]

Build FAILED.

/home/dependabot/.nuget/packages/gitversion.msbuild/5.12.0/tools/GitVersion.MsBuild.targets(9,9): error MSB3073: The command "dotnet --roll-forward Major "/home/dependabot/.nuget/packages/gitversion.msbuild/5.12.0/tools/netcoreapp3.1/gitversion.dll" "/home/dependabot/dependabot-updater/repo/ThScoreFileConverter.Core"  -output file -outputfile obj/gitversion.json" exited with code 1. [/home/dependabot/dependabot-updater/repo/ThScoreFileConverter.Core/ThScoreFileConverter.Core.csproj::TargetFramework=net8.0-windows]
    0 Warning(s)
    1 Error(s)

Time Elapsed 00:00:01.29


STDERR:


```

I investigated and found that the root cause is a mismatch in the following points:
* GitVersion requests an unshallow clone to calculate a SemVer version properly.
  Doc: https://gitversion.net/docs/reference/requirements#unshallow
* Dependabot always creates a shallow clone.
  Code: https://github.com/dependabot/dependabot-core/blob/c4c602fb9e6b636b6c4fa72181db60a070ee53bc/nuget/helpers/lib/NuGetUpdater/NuGetUpdater.Core/Clone/CloneWorker.cs#L122
  CI Log: https://github.com/y-iihoshi/ThScoreFileConverter/actions/runs/16914398911/job/47924452673#step:3:613
  ```
  updater | 2025/08/12 16:09:39 INFO Running command: git clone --no-tags --depth 1 --recurse-submodules --shallow-submodules https://github.com/y-iihoshi/ThScoreFileConverter /home/dependabot/dependabot-updater/repo
  ```

--- a/ThScoreFileConverter.Core/ThScoreFileConverter.Core.csproj
+++ b/ThScoreFileConverter.Core/ThScoreFileConverter.Core.csproj
@@ -16,15 +16,18 @@
 
   <ItemGroup>
     <PackageReference Include="CommunityToolkit.Diagnostics" Version="8.4.0" />
-    <PackageReference Include="GitVersion.MsBuild" Version="5.12.0">
-      <PrivateAssets>all</PrivateAssets>
-      <IncludeAssets>runtime; build; native; contentfiles; analyzers; buildtransitive</IncludeAssets>
-    </PackageReference>
     <PackageReference Include="IDisposableAnalyzers" Version="4.0.8">
       <PrivateAssets>all</PrivateAssets>
       <IncludeAssets>runtime; build; native; contentfiles; analyzers; buildtransitive</IncludeAssets>
     </PackageReference>
     <PackageReference Include="StyleCop.Analyzers.Unstable" Version="1.2.0.556">
+      <PrivateAssets>all</PrivateAssets>
+      <IncludeAssets>runtime; build; native; contentfiles; analyzers; buildtransitive</IncludeAssets>
+    </PackageReference>
+  </ItemGroup>
+
+  <ItemGroup Condition="'$(DEPENDABOT_JOB_ID)' == ''">
+    <PackageReference Include="GitVersion.MsBuild" Version="5.12.0">
       <PrivateAssets>all</PrivateAssets>
       <IncludeAssets>runtime; build; native; contentfiles; analyzers; buildtransitive</IncludeAssets>
     </PackageReference>

--- a/ThScoreFileConverter/ThScoreFileConverter.csproj
+++ b/ThScoreFileConverter/ThScoreFileConverter.csproj
@@ -54,10 +54,6 @@
       <PrivateAssets>all</PrivateAssets>
       <IncludeAssets>runtime; build; native; contentfiles; analyzers; buildtransitive</IncludeAssets>
     </PackageReference>
-    <PackageReference Include="GitVersion.MsBuild" Version="5.12.0">
-      <PrivateAssets>all</PrivateAssets>
-      <IncludeAssets>runtime; build; native; contentfiles; analyzers; buildtransitive</IncludeAssets>
-    </PackageReference>
     <PackageReference Include="IDisposableAnalyzers" Version="4.0.8">
       <IncludeAssets>runtime; build; native; contentfiles; analyzers</IncludeAssets>
       <PrivateAssets>all</PrivateAssets>
@@ -74,6 +70,13 @@
       <PrivateAssets>all</PrivateAssets>
     </PackageReference>
     <PackageReference Include="WPFLocalizeExtension" Version="3.10.0" />
+  </ItemGroup>
+
+  <ItemGroup Condition="'$(DEPENDABOT_JOB_ID)' == ''">
+    <PackageReference Include="GitVersion.MsBuild" Version="5.12.0">
+      <PrivateAssets>all</PrivateAssets>
+      <IncludeAssets>runtime; build; native; contentfiles; analyzers; buildtransitive</IncludeAssets>
+    </PackageReference>
   </ItemGroup>
 
   <ItemGroup>

